src/mpi/init/init_async.c

```diff
@@ -59,22 +74,163 @@ static void progress_fn(void *data)
     return;
 }
 
+#if defined(MPL_HAVE_MISC_GETNPROCS)
```

Comment:
How about --
```
#ifdef MPL_HAVE_MISC_GETNPROCS
#define DO_ASYNC_THREAD_AFFINITY
#endif
```
Then use `#ifdef DO_ASYNC_THREAD_AFFINITY` for the better readability?

Reply:
If `DO_ASYNC_THREAD_AFFINITY` is defined in mpl_misc.h, does it need to start with `MPL_`?

Reply:
Don't define it in `mpl`. Define it at the top of `init_async.c`

Reply:
Done